### PR TITLE
[NO GBP] Fixes airlock cycle links not working

### LINKED ISSUE #65882: airlock cyclelink helpers seem to not work
## Round ID:
[180896](https://scrubby.melonmesa.com/round/180896) , 180906

## Testmerges:
many april fools ones, check scrubby or statbus i guess

## Reproduction:
open a set of airlocks (brig entrance (new icebox brig) or science entrance (on tram) are a notable examples), notice airlocks dont cycle. examine airlock ingame and see it has matching cyclelink id, open sdmm and check if cyclelinks are properly put in the map,  be clueless if they are properly placed, dont know how to fix it

this also could be caused by something april fools related

### PR DESCRIPTION
**Pull request by dragomagol**
## About The Pull Request
#65580 , famous last words.
image

It turns out that airlocks apply most helpers on themselves, and that's all done in LateInitialize, so other airlock helpers (notably cycle links) were getting btfo'd.

~~At current, the solution I've created only works on the 1 directional cycle links, still have to figure out what's up with the multi cycle links.~~ Tested it, it works for both (and for access helpers).

This moves some of the code for applying helpers over to the helper code, reverts the previous change of moving payload to LateInitialize, and makes the access application process a little more sane.

## Why It's Good For The Game
Cycle links should work!
Fixes #65882

## Changelog

:cl:
fix: Airlock cycle links work now!
/:cl:

### CODE REVIEW

**Comment by capsaicinz** [comment]
this should fix #65882

File: code/modules/mapping/mapping_helpers.dm [hunk 1]
```diff
@@ -107,15 +107,50 @@
...
+	if(airlock.cyclelinkeddir)
+		airlock.cyclelinkairlock()
+	if(airlock.closeOtherId)
+		airlock.update_other_id()
+	if(airlock.abandoned)
+		var/outcome = rand(1,100)
+		switch(outcome)
+			if(1 to 9)
+				var/turf/here = get_turf(src)
+				for(var/turf/closed/T in range(2, src))
```
Comment (SuperNovaa41) [inline code/modules/mapping/mapping_helpers.dm:133]: die die die

File: code/modules/mapping/mapping_helpers.dm [hunk 2]
```diff
@@ -107,15 +107,50 @@
 	. = ..()
 	if(!mapload)
 		log_mapping("[src] spawned outside of mapload!")
-		return INITIALIZE_HINT_QDEL
+		return
```
Comment (AnturK) [inline code/modules/mapping/mapping_helpers.dm:110]: Why was this removed ? This is still invalid use right ?

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/mapping/mapping_helpers.dm (hunk 2 above)
Comment (dragomagol) [reply code/modules/mapping/mapping_helpers.dm:110]: We still have this in code:
https://github.com/tgstation/tgstation/blob/cbdcefefea65fe27ab9e9c00f7c1f5e826d8524d/code/modules/mapping/mapping_helpers.dm#L96-L98
If we have `. = ..()` and that covers what we should be returning, should we `return .` ?

File: code/modules/mapping/mapping_helpers.dm (hunk 1 above)
Comment (dragomagol) [reply code/modules/mapping/mapping_helpers.dm:133]: !(image)[https://cdn.discordapp.com/emojis/845166929929043989.webp?size=96&quality=lossless]

File: code/modules/mapping/mapping_helpers.dm (hunk 2 above)
Comment (AnturK) [reply code/modules/mapping/mapping_helpers.dm:110]: No this is fine in that case, though arguably making it explicit is easier to read.
